devices/Sen5x/Sen5xSensor.cs

```diff
@@ -0,0 +1,355 @@
+﻿// Copyright (c) 2017 The nanoFramework project contributors
+// See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Threading;
+using Iot.Device.Sen5x.Entities;
+using Iot.Device.Sen5x.Extensions;
+
+namespace Iot.Device.Sen5x
+{
+    /// <summary>
+    /// Documentation for this public class.
+    /// </summary>
+    public class Sen5xSensor
+    {
+        /// <summary>
+        /// The I2C address as specified in the datasheet.
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x69;
+
+        /// <summary>
+        /// The I2C bus speed as specified in the datasheet.
+        /// </summary>
+        public const I2cBusSpeed DefaultI2cBusSpeed = I2cBusSpeed.StandardMode;
+
+        private I2cDevice _i2c;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Sen5xSensor"/> class.
+        /// </summary>
+        /// <param name="i2cDevice">The I2cDevice to which this sensor is connected.</param>
+        public Sen5xSensor(I2cDevice i2cDevice)
+        {
+            _i2c = i2cDevice;
+        }
+
+        /// <summary>
+        /// Starts the measurement. After power up, the module is in Idle-Mode. Before any measurement values can be read, the Measurement-Mode needs to be started using this command.
+        /// </summary>
+        public void StartMeasurement()
+        {
+            Read(0x0021, TimeSpan.FromMilliseconds(50));
```

Comment:
rather than all those magic numbers, can you create an internal enum with a proper naming for them?
It's much nicer and easier to maintain. It does avoid problems later on when adding a new one for example.
So change as well the signature of your Read function to take this enum and things will be easier to maintain